x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/index.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { withApmSpan } from '../../../../utils/with_apm_span';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { getDuration } from './duration';
+import { getColdStartDuration } from './cold_start_duration';
+import { getMemoryChartData } from '../shared/memory';
+import { getComputeUsage } from './compute_usage';
+import { getActiveInstances } from './active_instances';
+import { getColdStartCount } from './cold_start_count';
+
+export function getServerlessAgentMetricCharts({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  serviceNodeName,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  serviceNodeName?: string;
+  start: number;
+  end: number;
+}) {
+  return withApmSpan('get_serverless_agent_metric_charts', async () => {
+    const options = {
+      environment,
+      kuery,
+      setup,
+      serviceName,
+      start,
+      end,
+    };
+    return await Promise.all([
+      getDuration({ ...options, serviceNodeName }),
+      getColdStartDuration({ ...options, serviceNodeName }),
+      getColdStartCount({ ...options, serviceNodeName }),
+      getMemoryChartData({ ...options, serviceNodeName }),
+      getComputeUsage({ ...options, serviceNodeName }),
+      getActiveInstances(options),
```

Comment:
Are we planning to show metrics per instance (`service.node.name`) like with Java? I notice we pass the `serviceNodeName` above but not here. 

Do we intend to re-use the endpoint `GET /internal/apm/services/{serviceName}/serviceNodes` for aws lambda metric instances? It's currently jvm but can be changed to support other agents.

Reply:
I don't see much value in showing the metrics per lambda instance. 
BUT, it's possible that an APM service groups multiple different AWS Lambda functions. In this case it would be nice to be able to display the metrics for individual Lambda functions (i.e. `faas.id` field)

Reply:
@AlexanderWert so what you want to show is the `cardinality` per `faas.id`?

Reply:
@cauemarcondes 
I meant it to be similar to what we have with JVM metrics today. There, `service.node.name` is an (optional) filter for the charts, right. So if one selects one specific JVM it shows the chart only for that JVM. But, if none specific JVM is selected it shows the aggregated chart over all JVMs.

The same should be possible here, but with `faas.id` instead of `service.node.name`. So, when no specific lambda function is selected then we show all the above charts as an aggregate over all lambda functions (within an APM service). But if the user selects one specific Lambda function (i.e. `faas.id`) then the above charts are filtered on that `faas.id`. Does it make sense?

Reply:
passing the `faas.id` now: https://github.com/elastic/kibana/pull/139041/commits/8f01c18f0e0bdb186282f7e20a0fa88f879945cb#diff-984926e21b42f221d677a056e78fd3d30960c8d110b306a4c85413c14604e893R42

Reply:
@AlexanderWert can `service.node.name` not be populated with the value from `faas.id`? (i'm not sure if they are semantically different)